src/game/Game.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+** EPITECH PROJECT, 2022
+** Bomberman
+** File description:
+** Game
+*/
+
+#include "Game.hpp"
+#include "ecs/resource/Timer.hpp"
+#include "logger/Logger.hpp"
+#include "raylib/core/scoped.hpp"
+
+#include "components/Collidable.hpp"
+#include "components/Controlable.hpp"
+#include "components/Cube.hpp"
+#include "components/CubeColor.hpp"
+#include "components/Destructible.hpp"
+#include "components/Living.hpp"
+#include "components/Position.hpp"
+#include "components/Velocity.hpp"
+
+#include "systems/ChangeCube.hpp"
+#include "systems/Collision.hpp"
+#include "systems/DrawingCube.hpp"
+#include "systems/InputManager.hpp"
+#include "systems/Movement.hpp"
+
+namespace game
+{
+    Game::Game(ecs::World &world, Parameters params)
+        : _world(world), _map(static_cast<size_t>(params.mapSize.x), static_cast<size_t>(params.mapSize.y)),
+          _params(params)
+    {
+    }
+
+    const map::Map &Game::getMap() const { return _map; }
+
+    void Game::setup(raylib::core::Camera3D &camera)
+    {
+        size_t width = _map.getWidth();
+        size_t depth = _map.getHeight();
+
+        camera.setPosition(
+            {width / 2.f, 8 /*static_cast<float>(width)*/, static_cast<float>(depth)}); // Camera position
+        camera.setTarget({width / 2.f, 0.f, depth / 2.f});                              // Camera looking at point
+        camera.setUp({0.0f, 1.0f, 0.0f}); // Camera up vector (rotation towards target)
+        camera.setFovY(75.0f);            // Camera field-of-view Y
+        camera.setProjection(CAMERA_PERSPECTIVE);
+
+        /// Add world ressources
+        _world.addResource<game::Users>();
+        _world.addResource<ecs::Timer>();
+        /// Add world systems
+        _world.addSystem<systems::InputManager>();
+        _world.addSystem<systems::ChangeCube>();
+        _world.addSystem<systems::DrawingCube>();
+        _world.addSystem<systems::Movement>();
+        _world.addSystem<systems::Collision>();
+
+        for (size_t i = 0; i < _params.playerCount; i++) {
+            User::UserId owner = static_cast<User::UserId>(i);
+            Vector2 cell = _map.getPlayerStartingPosition(owner);
+
+            _world.addEntity()
+                .with<components::Position>(cell.x, 1.f, cell.y)
+                .with<components::Velocity>()
+                .with<components::Living>(_params.livesCount)
+                .with<components::Collidable>()
+                .with<components::Player>()
+                .with<components::Cube>()
+                .with<components::Size>(0.7f, 2.f, 0.7f)
+                .with<components::CubeColor>(raylib::core::Color::RED)
+                .with<components::Controlable>(owner,
+                    [this](ecs::Entity self, ecs::SystemData data, const Users::ActionEvent &event) {
```

Comment:
Moving this huge lambda into a separate function would make this a lot more readable.
Something like: `.with<...>(owner, &World::processEntityAction)`

Reply:
I have planned to do this for the next part of the game implementation.